packages/data-provider/src/repository/submittedRepository.ts

```diff
@@ -87,17 +87,24 @@ const repository = (dependencies: BaseDependencies) => {
 		 * Find SubmittedData by category ID and Organization with pagination
 		 * @param {number} categoryId Category ID
 		 * @param {string} organization Organization Name
+		 * @param {SQL} filter Optional filter
+		 * @param {paginationOps} paginationOps Pagination properties
 		 * @returns The SubmittedData found
 		 */
 		getSubmittedDataByCategoryIdAndOrganizationPaginated: async (
 			categoryId: number,
 			organization: string,
 			paginationOps: paginationOps,
+			filter?: SQL,
```

Comment:
This function is reusable by adding an optional SQL parameter.

Reply:
silly question: Are we using SQL in some endpoints and SQON in others? it's unclear to me what makes that difference necessary here.

Reply:
SQON is a JSON format input by the user in the request. This requires to be transformed to a format understandable by repository layer here is when we use `SQL` that is a type specific for `Drizzle ORM` library.